src/components/navigation/UserLink/UserLink.tsx

```diff
@@ -1,21 +1,40 @@
 import {
+  ActionIcon,
   Avatar,
-  Text,
-  useMantineTheme,
   Box,
+  Group,
   Space,
+  Text,
   UnstyledButton,
-  Group,
-  ActionIcon
+  useMantineTheme
 } from '@mantine/core'
 import Link from 'next/link'
 import { TbLogout } from 'react-icons/tb'
 import { routes } from '@config/routes'
-import { useFetchCurrentUser } from '@api/hooks/useFetchCurrentUser'
+import { useLogoutUser } from '@api/hooks/useLogoutUser'
+import { useRouter } from 'next/router'
+import { useContext } from 'react'
+import { noAuthRoute } from '@features/auth/utils/noAuthRoute'
+import { UserContext } from '@contexts/UserContext'
 
 export const UserLink = () => {
+  const router = useRouter()
   const theme = useMantineTheme()
-  const { username, email, avatarUrl } = useFetchCurrentUser()
+
+  const { mutate } = useLogoutUser()
+
+  const { reloadUser, user } = useContext(UserContext)
+
+  const { username, email, avatarUrl } = user
+
+  const handleLogout = () => {
+    mutate(undefined, {
```

Comment:
It's a convention to use null when you use it explicitly. Undefined typically means it's empty by default.

Reply:
When I use null, it throws an error